drivers/QSPI.cpp

```diff
@@ -13,223 +13,229 @@
-                     int address_width, int address_size,  
-                     int alt_width, int alt_size,
-                     int data_width,
-                     int dummy_cycles,
-                     int mode ) {
-    if(!IS_BUS_WIDTH_VALID(inst_width)) return false;
-    if(!IS_BUS_WIDTH_VALID(address_width)) return false;
-    if(!IS_SIZE_VALID(address_size)) return false;
-    if(!IS_BUS_WIDTH_VALID(alt_width)) return false;
-    if(!IS_ALT_SIZE_VALID(alt_size)) return false;
-    if(!IS_BUS_WIDTH_VALID(data_width)) return false;
-    if(dummy_cycles < 0) return false;                         
-    if(mode != 0 && mode != 1) return false;                            
+qspi_return_status_t QSPI::configure_format(qspi_config_bus_width_t inst_width, qspi_config_bus_width_t address_width, qspi_config_address_size_t address_size, qspi_config_bus_width_t alt_width, qspi_config_alt_size_t alt_size, qspi_config_bus_width_t data_width, int dummy_cycles, int mode ) {
+    if(!IS_BUS_WIDTH_VALID(inst_width)) return QSPI_INVALID_PARAMETER;
+    if(!IS_BUS_WIDTH_VALID(address_width)) return QSPI_INVALID_PARAMETER;
+    if(!IS_SIZE_VALID(address_size)) return QSPI_INVALID_PARAMETER;
+    if(!IS_BUS_WIDTH_VALID(alt_width)) return QSPI_INVALID_PARAMETER;
+    if(!IS_ALT_SIZE_VALID(alt_size)) return QSPI_INVALID_PARAMETER;
+    if(!IS_BUS_WIDTH_VALID(data_width)) return QSPI_INVALID_PARAMETER;
+    if(dummy_cycles < 0) return QSPI_INVALID_PARAMETER;                         
+    if(mode != 0 && mode != 1) return QSPI_INVALID_PARAMETER;                            
     
     lock();
     switch(inst_width) {
-        case 1:_inst_width = QSPI_CFG_BUS_SINGLE; break;
-        case 2:_inst_width = QSPI_CFG_BUS_DUAL; break;
-        case 4:_inst_width = QSPI_CFG_BUS_QUAD; break;            
+        case QSPI_BUS_SINGLE:_inst_width = QSPI_CFG_BUS_SINGLE; break;
+        case QSPI_BUS_DUAL:_inst_width = QSPI_CFG_BUS_DUAL; break;
+        case QSPI_BUS_QUAD:_inst_width = QSPI_CFG_BUS_QUAD; break;            
         default:_inst_width = QSPI_CFG_BUS_SINGLE;
     }
     
     switch(address_width) {
-        case 1:_address_width = QSPI_CFG_BUS_SINGLE; break;
-        case 2:_address_width = QSPI_CFG_BUS_DUAL; break;
-        case 4:_address_width = QSPI_CFG_BUS_QUAD; break;            
+        case QSPI_BUS_SINGLE:_address_width = QSPI_CFG_BUS_SINGLE; break;
+        case QSPI_BUS_DUAL:_address_width = QSPI_CFG_BUS_DUAL; break;
+        case QSPI_BUS_QUAD:_address_width = QSPI_CFG_BUS_QUAD; break;            
         default:_address_width = QSPI_CFG_BUS_SINGLE;
     }
     
     switch(address_size) {
-        case 8:_address_size = QSPI_CFG_ADDR_SIZE_8; break;
-        case 16:_address_size = QSPI_CFG_ADDR_SIZE_16; break;
-        case 24:_address_size = QSPI_CFG_ADDR_SIZE_24; break;
-        case 32:_address_size = QSPI_CFG_ADDR_SIZE_32; break;            
+        case QSPI_ADDR_SIZE_8:_address_size = QSPI_CFG_ADDR_SIZE_8; break;
```

Comment:
each statement on own line please, same for if () above on lines 52 and below

Reply:
Will fix.